cost_rules/app.py

```diff
@@ -0,0 +1,263 @@
+                'DimensionName': 'TAG',
+                'DimensionValue': tag,
+            }
+        }
+        for tag in tag_names
+    ]
+
+    return results
+
+
+def build_rules(chart_codes, tag_names, account_codes):
+    '''
+    Build a list of cost-category rules.
+    Rule order matters, first rule matched wins.
+
+    For each program code in our chart of accounts, create rules that:
+      1. Check resource tags for the program code.
+      2. Check for accounts tagged with the program code.
+    Finally, if an unmatched tag value exists, create a new category from it.
+
+    '''
+    rules = []
+
+    # first, generate rules for each program code
+    for code, name in chart_codes.items():
+        safe_name = _strip_special_chars(name)
+
+        if safe_name != name:
+            LOG.info(f'{name} renamed to {safe_name}')
+
+        title = f"{code} {safe_name}"
+
+        # generate rules checking each tag
+        rules.extend(_build_tag_rules(
+            title,
+            tag_names,
+            code
+        ))
+
+        if code in account_codes:
+            # if any accounts are tagged with this code,
+            # add a rule for them here
+            rules.append(_build_account_rule(
+                title,
+                tag_names,
+                account_codes[code]
+            ))
+
+    # finally, inherit tag values if no other rule matched
+    rules.extend(_build_inherit_rules(tag_names))
+
+    return rules
+
+
+def lambda_handler(event, context):
+    """Sample pure Lambda function
+
+    Parameters
+    ----------
+    event: dict, required
+        API Gateway Lambda Proxy Input Format
+
+        Event doc: https://docs.aws.amazon.com/apigateway/latest/developerguide/set-up-lambda-proxy-integrations.html#api-gateway-simple-proxy-for-lambda-input-format
+
+    context: object, required
+        Lambda Context runtime methods and attributes
+
+        Context doc: https://docs.aws.amazon.com/lambda/latest/dg/python-context-object.html
+
+    Returns
+    ------
+    API Gateway Lambda Proxy Output Format: dict
+
+        Return doc: https://docs.aws.amazon.com/apigateway/latest/developerguide/set-up-lambda-proxy-integrations.html
+    """
+
+    result = { "statusCode": 200 }
+    try:
+        # get environment variables
+        chart_url = _get_os_var('ChartOfAccountsURL')
+
+        _tag_list = _get_os_var('TagList')
+        tag_list = _parse_env_list(_tag_list)
+
+        # get account tags
+        account_codes = collect_account_tag_codes(tag_list)
+
+        # get chart of accounts
+        chart_json = requests.get(chart_url)
+        chart_json.raise_for_status()
+        chart_data = chart_json.json()
+
+        # generate rules
+        rules_data = build_rules(chart_data, tag_list, account_codes)
+
+        # return as json string
+        result["body"] = json.dumps(rules_data)
+        result["headers"] = { "content-type": "application/json; charset=utf-8" }
```

Comment:
hmm, the `rules` parameter of `AWS::CE::CostCategory` takes a string so should this maybe return a `text/plain` string?  otherwise we would need a way to convert json to a string to pass to `rules` parameter

Reply:
sorry, i was confused.  rest response header has nothing to do with how the result is passed to sceptre params.  leave it as-is is fine.